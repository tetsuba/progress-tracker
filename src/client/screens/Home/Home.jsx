import React from 'react'

import { Col, Container, Row } from 'react-bootstrap'
import { Link } from 'react-router-dom'

export default function Home() {
  return (
    <Container>
      <Row className="mt-5">
        <Col>
          <h1>Welcome to Progress Tracker...</h1>
        </Col>
      </Row>
      <Row className="mt-5">
        <Col sm={4}>
          <Link to="/students" className="card text-body text-decoration-none">
            <div className="card-header">
              <h1>Students</h1>
            </div>
            <div className="card-body">Students</div>
            <div className="card-footer">Number of students</div>
          </Link>
        </Col>

        <Col sm={4}>
          <Link to="/myAccount" className="card text-body text-decoration-none">
            <div className="card-header">
              <h1>My Account</h1>
            </div>
            <div className="card-body">My Account</div>
            <div className="card-footer"></div>
          </Link>
        </Col>
      </Row>
    </Container>
  )
}
