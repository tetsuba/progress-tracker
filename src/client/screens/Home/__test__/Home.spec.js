import { render } from '@testing-library/react'
import Home from '../Home'
import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { BrowserRouter as Router } from 'react-router-dom'
// import ReactTestUtils from 'react-dom/test-utils';
import renderer from 'react-test-renderer'

describe.skip('@Home', () => {
  // const {getByText} =  render(<Login />);
  // console.log('root', root)

  it('renders default', () => {
    // const { getByText, debug } = renderer(<Router><Home /></Router>);
    // debug()
    // expect(getByText('Welcome to Progress Tracker...')).toMatchSnapshot();

    const tree = renderer
      .create(
        <Router>
          <Home />
        </Router>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
