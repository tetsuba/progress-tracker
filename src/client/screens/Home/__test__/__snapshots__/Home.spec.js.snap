// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@Home renders default 1`] = `
<div
  className="container"
>
  <div
    className="mt-5 row"
  >
    <div
      className="col"
    >
      <h1>
        Welcome to Progress Tracker...
      </h1>
    </div>
  </div>
  <div
    className="mt-5 row"
  >
    <div
      className="col-sm-4"
    >
      <a
        className="card text-body text-decoration-none"
        href="/students"
        onClick={[Function]}
      >
        <div
          className="card-header"
        >
          <h1>
            Students
          </h1>
        </div>
        <div
          className="card-body"
        >
          Students
        </div>
        <div
          className="card-footer"
        >
          Number of students
        </div>
      </a>
    </div>
    <div
      className="col-sm-4"
    >
      <a
        className="card text-body text-decoration-none"
        href="/myAccount"
        onClick={[Function]}
      >
        <div
          className="card-header"
        >
          <h1>
            My Account
          </h1>
        </div>
        <div
          className="card-body"
        >
          My Account
        </div>
        <div
          className="card-footer"
        />
      </a>
    </div>
  </div>
</div>
`;
